source/extensions/filters/http/decompressor/decompressor_filter.h

```diff
@@ -100,9 +100,18 @@ class DecompressorFilterConfig {
   const std::string& contentEncoding() { return decompressor_factory_->contentEncoding(); }
   const RequestDirectionConfig& requestDirectionConfig() { return request_direction_config_; }
   const ResponseDirectionConfig& responseDirectionConfig() { return response_direction_config_; }
+  // TODO: replace std::vector for std::pair. For some reason the macro is failing to build with a
+  // multivariate templated type.
+  const std::vector<Http::LowerCaseString>& trailersStrings() const {
+    CONSTRUCT_ON_FIRST_USE(
+        std::vector<Http::LowerCaseString>,
+        {Http::LowerCaseString(fmt::format("{}-compressed-bytes", trailers_prefix_)),
+         Http::LowerCaseString(fmt::format("{}-uncompressed-bytes", trailers_prefix_))});
+  }
```

Comment:
What are your thoughts on using 2 separate values rather than relying on indices of strings in this list in `decompressor_filter.cc`?

Reply:
Yep, that is what I want but the template is not compiling correctly with the macro being used here per my TODO above.